### ISSUE TITLE
5 user limit when using the public Jitsi Meet server

### ISSUE THREAD

**Issue by LordofDice** [luvolondon/fvtt-module-jitsiwebrtc#92]
In our last session only 5 out of 6 players could use the audio / video chat. We relogged and restarted the server several times and always the last one that came in could hear and see all others but was not able to speak or send video (but he could see himself in the video on his own PC). The User with the problem was always the last one, so we're sure it wasn't a local audio / webcam problem. Is there a new limit of 5 users? In the past we had no problems with 6 users.

**Comment by ravingdragoon** [comment]
I am having the same issue

**Comment by ravingdragoon** [comment]
Additionally, I am receiving the following every few seconds in the dev console

```
08:51:47.774  2021-12-05T16:51:37.781Z [modules/xmpp/strophe.ping.js] Ping timeout null
r @ Logger.js:154 
(anonymous) @ strophe.ping.js:133
Promise.then (async)
ping @ strophe.ping.js:90
(anonymous) @ strophe.ping.js:120
```
When r @ Logger.js:154 is hovered I see "webpack://JitsiMeetJS/.../@jitsi/logger/lib/logger.js:154"

**Comment by bekriebel** [comment]
There was no limit added on the module's end. Are you all using your own Jitsi servers, or are you using the public one? It's possible that they implemented a limit on the public server.

**Comment by LordofDice** [comment]
I just use the modul as addon for foundry Vtt. So  I'm not hosting my own Jitsi server (or at least I wouldn't be aware of it). 

We didn't have any problems last Saturday (28/11) so I guess something must have happened during the week.

**Comment by ravingdragoon** [comment]
Same. I use the module as is, I don't have a custom server.

**Comment by ravingdragoon** [comment]
Any idea what would cause the error I noted?

**Comment by LordofDice** [comment]
What I have done now is reset the Jitsi Meet Room ID in the Modulsettings. But I do not know if it has any effect, because I don't have six people there for testing at the moment.

**Comment by ravingdragoon** [comment]
I tried that last night, with no success..

**Comment by bekriebel** [comment]
This appears to be a new limit that has been imposed by the public Jitsi Meet server. Even if I go directly to https://beta.meet.jit.si and use their web interface, I'm not able to get more than 5 users to display at the same time.

Unfortunately, there's not really anything I can do about this. The best alternative is likely to run your own Jitsi sever without this restriction. However, if you go down that path, I would highly recommend instead using my [LiveKit AVClient](https://foundryvtt.com/packages/avclient-livekit) module instead.

Issues like this are why I started developing a new AV module a few months ago. Jitsi has always been kind of a pain for me to work with. I started evaluating other options recently and landed on LiveKit as a new option. It's much nicer to work with from the development side, and from a user standpoint it has been MUCH more stable than Jitsi ever was for me. The downside is that there are no free public LiveKit servers. I've provided links and even my own tutorial on how to run your own LiveKit server. I also run a highly available LiveKit cluster with nodes throughout the world. However, I unfortunately can't afford to run this cluster for free and do need to charge for it, so it's only available to subscribers of my Patreon.

One other thing that you can also attempt to do is to go into the Audio/Video Configuration settings under the Server tab, switch to `Custom Server` and put in the address `meet.jit.si`. This is the non-beta version of Jitsi's public server. However, this isn't used by default because many people had connection issues when using it. Also, if they put the restriction on the beta server recently, I suspect that this version will also get restricted soon.

I wish there was more that I could do, but this was always a risk with Jitsi's public servers and it's unfortunately out of my hands. I know it's either more work or costs some money, but I do think your experience will be much better if you upgrade to the LiveKit module.

![image](https://user-images.githubusercontent.com/183779/144762768-945df310-c031-428c-880d-25008909e31a.png)

**Comment by marmitejim** [comment]
Thanks for the open, honest answer to this bekriebel. I got the same issue today.
Happy to support your work so I'll bob over to your Patreon and get your LiveKit alternative.
We are based in the UK. Should be ok for us yeah?

**Comment by bekriebel** [comment]
> Thanks for the open, honest answer to this bekriebel. I got the same issue today.
> 
> Happy to support your work so I'll bob over to your Patreon and get your LiveKit alternative.
> 
> We are based in the UK. Should be ok for us yeah?

You're very welcome, and thanks for the support!

Yes, one of the nodes is in Frankfurt, Germany which I believe should give you decently low latency. I keep an eye on the cluster and add nodes as needed. It will automatically route to the closest node to the first person to join your game.

**Comment by LordofDice** [comment]
Then my suspicions were confirmed. Thanks for checking out and the great job! 

We are testing the on-board Foundry A / V resources first. When it works okay, otherwise we will use the Live Kit.

**Comment by pgillett** [comment]
> This appears to be a new limit that has been imposed by the public Jitsi Meet server. Even if I go directly to https://beta.meet.jit.si and use their web interface, I'm not able to get more than 5 users to display at the same time.

Have you got a reference to the 5 user limit? After struggling yesterday, we switched over to a separate (public) Jitsi meeting and connected 7 fine.

**Comment by bekriebel** [comment]
> Have you got a reference to the 5 user limit? After struggling yesterday, we switched over to a separate (public) Jitsi meeting and connected 7 fine.

I'm just basing it on manual testing that I did. After confirming the issue when using the module, I attempted to load more than 5 users directly using their web interface. In multiple attempts, as soon as more than 5 users were joined, the subsequent users weren't able to share video.

If it worked after retrying, perhaps they are running multiple nodes for their server and not all have the limit. There's also always the chance that they will revert the change.

**Comment by pgillett** [comment]
I've just done a test (directly in the browser) with meet.jit.si and can add more than 5 video streams. Using beta.meet.jit.si disables above 5

**Comment by bekriebel** [comment]
I just tested again and found that in the 6th user, Jitsi's API is actually spawning an error ``Source add rejected. Maximum number of video senders reached.`
![image](https://user-images.githubusercontent.com/183779/144898708-92113127-8931-4f0b-8402-a556843cce65.png)

This let me track down where it is in the code: https://github.com/jitsi/jicofo/pull/827. As expected, it's a server-side setting than is configurable per instance.

I've updated the README with information about this and suggestions of trying the alternate address or switching to LiveKit: https://github.com/luvolondon/fvtt-module-jitsiwebrtc#user-limit-warning

**Comment by bekriebel** [comment]
> I've just done a test (directly in the browser) with meet.jit.si and can add more than 5 video streams. Using beta.meet.jit.si disables above 5

Yep. We'll see if they implement the same restriction on the non-beta site soon. It looks like the released version that includes this code change just dropped today. Also, in the past when I set the default server to be `meet.jit.si`, I started getting several reports of connection issues. For some reason, it seems like several clients have issues using that version, which is why the module has used the beta address for so long.
